src/NovelCoreClrService.cpp

```diff
@@ -0,0 +1,130 @@
+//
+// Created by tagoo on 19/09/2019.
+//
+
+#include <cassert>
+#include <iostream>
+#include <vector>
+#include <coreclr_delegates.h>
+#include <hostfxr.h>
+#include <nethost.h>
+#include "NovelCoreClrService.h"
+
+#if defined(_WIN32)
+  #include <Windows.h>
+#else
+  #include <dlfcn.h>
+#endif
+
+const auto HostApiBufferTooSmall = 0x80008098;
+
+static void* getNativeExport(void* nativeLibrary, const char* exportName) {
+#if defined(_WIN32)
```

Comment:
Need to have some logic for loading and finding the hostfxr exports that differs between Windows and Unix.